versioned/persist/store/src/main/java/org/projectnessie/versioned/persist/store/PersistVersionStore.java

```diff
@@ -111,44 +112,67 @@ public Hash commit(
     for (Operation<CONTENT> operation : operations) {
       if (operation instanceof Put) {
         Put<CONTENT> op = (Put<CONTENT>) operation;
-        ContentId contentId = ContentId.of(storeWorker.getId(op.getValue()));
+        CONTENT content = op.getValue();
+        CONTENT expected = op.getExpectedValue();
+
+        if (storeWorker.getId(content) == null) {
+          // No content-ID --> New content
+
+          if (expected != null) {
+            throw new IllegalArgumentException(
+                String.format(
+                    "Content id is null, which means 'create new content', "
+                        + "but expected content is not null, which means modifying "
+                        + "an existing content - key '%s'",
+                    op.getKey()));
+          }
+
+          // assign content-ID
+          content = storeWorker.applyId(content, UUID.randomUUID().toString());
+        }
+
+        ContentId contentId = ContentId.of(storeWorker.getId(content));
         commitAttempt.addPuts(
             KeyWithBytes.of(
                 op.getKey(),
                 contentId,
-                storeWorker.getPayload(op.getValue()),
-                storeWorker.toStoreOnReferenceState(op.getValue())));
+                storeWorker.getPayload(content),
+                storeWorker.toStoreOnReferenceState(content)));
 
-        if (op.getExpectedValue() != null) {
-          ContentId expectedContentId = ContentId.of(storeWorker.getId(op.getExpectedValue()));
+        if (expected != null) {
+          String expectedId = storeWorker.getId(expected);
+          if (expectedId == null) {
```

Comment:
maybe just use `Preconditions.checkArgument()`?

Reply:
We don't have Guava in this module yet - and all other places in this module do it like this.